linux/Dockerfile

```diff
@@ -19,6 +21,38 @@ RUN apt-get update && \
 	curl \
 	jq
 
+# --- kcov --- #
+# Install dependencies
+RUN apt-get install -y \
+	cmake \
+	libcurl4-openssl-dev \
+	libelf-dev \
+	libdw-dev \
+	gcc \
+	binutils-dev \
+	zlib1g-dev \
+	libiberty-dev \
+	pkg-config \
+	python2.7 \
+	wget \
+	zip
+
+# For some reason `python` isn't the default executable file name.
+# Needed for kcov
+RUN ln -s /usr/bin/python2.7 /usr/bin/python
+
+# Download kcov
+# -nc is okay since we are downloading a tagged version
+RUN wget -nc https://github.com/SimonKagstrom/kcov/archive/v$KCOV_VERSION.zip
+RUN unzip -uoq v$KCOV_VERSION.zip
+
+# Build KCov
+RUN if ! test -d "kcov-$KCOV_VERSION/${KCOV_BUILD_DIR}"; then mkdir "kcov-$KCOV_VERSION/${KCOV_BUILD_DIR}" ; fi
+RUN cd "kcov-$KCOV_VERSION/${KCOV_BUILD_DIR}" && \
```

Comment:
I would combine this with above. Also no need to test before mkdir, just use `-p`.

Additionaly, I would also try removing the build dir after `make install`, as it should no longer be required. If this fails, then just drop the `rm`.

```dockerfile
RUN mkdir -p "kcov-$KCOV_VERSION/${KCOV_BUILD_DIR}" && \
    cd "kcov-$KCOV_VERSION/${KCOV_BUILD_DIR}" && \ 
	cmake .. -DCMAKE_INSTALL_PREFIX=/usr/local -DCMAKE_BUILD_TYPE=Release && \
	make && \
	make install && \
    rm -rf "kcov-$KCOV_VERSION/${KCOV_BUILD_DIR}"
```